plugins/modules/lvol.py

```diff
@@ -236,12 +262,118 @@ def create_lv(module, name):
     lv_run_cmd(module, cmd, success_msg, fail_msg, None)
 
 
-def modify_lv(module, name):
+def extend_lv(module, name, init_props):
+    """
+    Extend a logical volume with the given new size.
+    arguments:
+        module (dict): The Ansible module
+        name (str): Logical Volume Name
+        init_props (str): Initial properties of the logical volume
+    note:
+        The new size must be large than the original size.
+    return:
+        none
+    """
+    global result
+
+    # get lvid to fetch additonal information
+    pattern = r"^LV IDENTIFIER:\s+(\w+\.\d+)"
+    lvid = re.search(pattern, init_props, re.MULTILINE).group(1)
+
+    # get the physical partition size (PP size) for converting
+    # size with prefixes B/b, K/k, M/m, and G/g into corresponding
+    # number of logical partitions
+    # also fetch the current lv size in logical partions (LPs)
+    # e.g output of lquerykv wit -cst
+    # Csize: <number of current LPs in LV>
+    # PPsize: <PP size of each LP>
+    cmd = "lquerylv -L %s -cst" % lvid
+    fail_msg = "Failed to fetch the physical partition size and current \
+        number of logical partitions in the logical volume"
+    lv_run_cmd(module, cmd, None, fail_msg, fetch=True)
+    pattern = r"^Csize:\s+(\d+)"
+    curr_lps_in_lv = int(re.search(pattern, result['stdout'], re.MULTILINE).group(1))
+    pattern = r"^PPsize:\s+(\d+)"
+    pp_size = int(re.search(pattern, result['stdout'], re.MULTILINE).group(1))
+
+    # check if plus (+) sign is used with the 'size' parameter
+    size = module.params['size']
+    plus_sign_used = False
+    if size[0] == "+":
+        plus_sign_used = True
+        size = size[1:]  # remove plus sign
+    elif size[0] == "-":
+        # minus sign is not supported
+        result['msg'] += "\nMinus sign (-) is not a supported prefix for 'size' parameter."
+        module.fail_json(**result)
+
+    # convert 'size' parameter to corresponding number of LPs
+    # needed to satisfy the specified new size if suffixes are
+    # used (B, b, K, k, M, m, G, g)
+    # calculation logic lifted verbatim from extendlv cmd in lvm
+    pp_size = 1 << pp_size
+    if (size[-1] == "G") or (size[-1] == "g"):
+        shift = 30
+    elif (size[-1] == "M") or (size[-1] == "m"):
+        shift = 20
+    elif (size[-1] == "K") or (size[-1] == "k"):
+        shift = 10
+    elif (size[-1] == "B") or (size[-1] == "b"):
+        shift = 9
+    else:  # no suffix
+        shift = 0
+
+    # conversion from size with suffix to correspoding num
```

Comment:
Why we need to calculate the size before extendlv is called?  don't exendlv command already support G, M, B, K?

Reply:
yes, extendlv does the calculation for those. i am doing it here because of the "+" sign feature.
extendlv does not support the "+" sign feature.

in lvm, extendlv is used to increase the size of an existing lv.
so if testlv has 5 mb size, then running extendlv 10M will add 10mb to testlv on top of
the 5mb

in this enhancement, using the "+" sign feature will produce the behavior described previously.

however, if we do not have "+" sign with the size on the parameter for this module on an existing lv
then, this feature will instead try to change the size of the existing lv to the specified size.

for this to happen, i need to do the calculations used inside extendlv for the suffixes (K, M, G, etc)
so i know how many PPs (physical partitions) is equivalent to the specified suffixed size, then 
i need to know the current PPs. the difference between the current and specified PPs, are what is
needed to extend the existing lv such that it will match the specified size (size parameter without using "+" sign)